features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -146,6 +146,16 @@ DWORD get_fattime(void)
            | (DWORD)(ptm->tm_sec/2    );
 }
 
+void *ff_memalloc(UINT size)
+{
+    return malloc(size);
```

Comment:
Actually I changed this to propogate the error up, it's used for allocating large file buffers, so it's not unreasonable to be propogated as an error code (complies to posix and matches existing use of ff_memalloc in the chanfs layer).